packages/router/__tests__/router-test.ts

```diff
@@ -11214,7 +11302,7 @@ describe("a router", () => {
         );
       });
 
-      it("should handle not found routes with a 404 Response", async () => {
+      describe("Errors with Status Codes", () => {
```

Comment:
View this section with whitespace hidden, but now instead of returning responses we return a new `ErrorWithStatus` instance and we align the error messages with what Remix uses